lib/models/player_relation.dart

```diff
@@ -4,6 +4,38 @@ import '../constants/constants.dart';
 import 'game.dart';
 import 'player.dart';
 
+enum RelationState {
+  mutual(acceptedByPlayer: true, acceptedByOther: true),
+  sent(acceptedByPlayer: true, acceptedByOther: false),
+  received(acceptedByPlayer: false, acceptedByOther: true),
+  random(acceptedByPlayer: false, acceptedByOther: false);
+
+  final bool acceptedByPlayer;
+  final bool acceptedByOther;
+
+  const RelationState(
+      {required this.acceptedByPlayer, required this.acceptedByOther});
+
+  factory RelationState.fromDbMap(Map<String, dynamic> map, bool isFirst) {
```

Comment:
Unfortunately, the dart enum doesnt allow to just call RelationState(bool1, bool2). Thats why we need this factory.